features/steps/steps.py

```diff
@@ -211,14 +239,59 @@ def instance_getter(i,representation_id, representation_type, negative=False):
         if condition(i, representation_id, representation_type):
             return i
 
+def strip_split(stmt, strp = ' ', splt = ' '):
+    return list(
+        map(str.lower, map(lambda s: s.strip(strp), stmt.split(splt)))
+    )
+
+def include_subtypes(stmt):
+    stmt = strip_split(stmt, strp = '[]')
+    if len(stmt) > 1 and 'subtypes' in stmt:
+        excluding_statements = ['without', 'not', 'excluding', 'no']
+        if len(set(stmt).intersection(set(excluding_statements))):
+            return False
+        else:
+            return True
+    else:
+        return True
+
+def map_state(values, fn):
+    if isinstance(values, (tuple, list)):
+        return type(values)(map_state(v, fn) for v in values)
+    else:
+        return fn(values)
+
+@given("An {entity_opt_stmt}")
+def step_impl(context, entity_opt_stmt):
+    entity = entity_opt_stmt.split()[0]
```

Comment:
I actually would really like the pyparsing approach here again. What do you think?

```
ifc_entity = Ifc\w+
grammar = ifc_entity + [ "with subtypes" | "without subtypes" ]
```

Reply:
^ but save this for later :)

Reply:
Sounds good :) It is also nice work to do.

Linked it to the related issue so I can easily find it back later https://github.com/buildingSMART/ifc-gherkin-rules/issues/31